parser/parser/test_taxonomy_parser.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+import taxonomy_parser as parser
+import re
+
+
+def test_normalized_filename():
+    x = parser.Parser()
+    normalizer = x.normalized_filename
+    name = normalizer("test")
+    assert name == "test.txt"
+    name = normalizer("test.txt")
+    assert name == "test.txt"
+    name = normalizer("t")
+    assert name == "t.txt"
+
+def test_fileiter():
+    x=parser.Parser()
+    file=x.file_iter("test.txt")
+    counter=0
+    language_code_prefix = re.compile('[a-zA-Z][a-zA-Z]:')
+    for _,line in file:
+        counter+=1
+        assert line == '' or line[0]=="#" or ":" in line
+        if counter==27:
+            assert line == "carbon_footprint_fr_foodges_value:fr:10"
+    assert counter == 37+1
```

Comment:
enumerate is your friend :-)

```suggestion
    language_code_prefix = re.compile('[a-zA-Z][a-zA-Z]:')
    for _,line, counter in enumerate(file):
        assert line == '' or line[0]=="#" or ":" in line
        if counter==26:
            assert line == "carbon_footprint_fr_foodges_value:fr:10"
    assert counter == 37
```